src/core/lib/event_engine/windows/event_poller.h

```diff
@@ -0,0 +1,100 @@
+// Copyright 2022 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef GRPC_CORE_LIB_EVENT_ENGINE_WINDOWS_EVENT_POLLER_H
+#define GRPC_CORE_LIB_EVENT_ENGINE_WINDOWS_EVENT_POLLER_H
+
+#include <grpc/support/port_platform.h>
+
+#ifdef GPR_WINDOWS
+
+// ---------------------------------------------------------------------------
+// TODO(hork): I forked this from the iomgr_engine due to some slight
+// differences in types. Both are being actively worked on, so the idea is to
+// develop in parallel and netogiate a reunification once things settle.
+//
+// This is most valuable if WrappedSocket/EventHandler and EventPoller can be
+// treated generally by the same TCP code. If not, unification might not be that
+// valuable.
+// ---------------------------------------------------------------------------
+
+
+#include <vector>
+
+#include "absl/status/status.h"
+#include "absl/strings/string_view.h"
+
+#include <grpc/event_engine/event_engine.h>
+
+#include "src/core/lib/gprpp/time.h"
+
+namespace grpc_event_engine {
+namespace experimental {
+
+class WrappedSocket {
+ public:
+  virtual SOCKET Socket() = 0;
+  // Shutdown a WrappedSocket. After this operation, NotifyXXX and SetXXX
+  // operations cannot be performed.
+  virtual void MaybeShutdown(absl::Status why) = 0;
+  // Schedule on_read to be invoked when the underlying socket
+  // becomes readable.
+  virtual void NotifyOnRead(absl::AnyInvocable<void()> on_read) = 0;
+  // Schedule on_write to be invoked when the underlying socket
+  // becomes writable.
+  virtual void NotifyOnWrite(absl::AnyInvocable<void()> on_write) = 0;
+  // Force set a readable event on the underlying socket.
+  virtual void SetReadable() = 0;
+  // Force set a writable event on the underlying socket.
+  virtual void SetWritable() = 0;
+  // Returns true if the WrappedSocket has been shutdown.
+  virtual bool IsShutdown() = 0;
+  virtual ~WrappedSocket() = default;
+};
+
+// A low-level poller that watches for socket events and dispatches
+// notifications.
+class EventPoller {
```

Comment:
Likely the split we want is:

class Poller {
 public:
  Shutdown(); 
  Work(Duration);
  Kick();
};

for posix: readable/writable notifications go into a PosixPoller type that we implement from each poller interface
for windows: IOCP just exports a Watch method, because we don't need the indirection

Reply:
Partially done. I broke out the interface for the Windows code, but will do the Posix pollers after the poll fork PR lands

Reply:
Resolved in https://github.com/grpc/grpc/pull/30413